src/js/services/integrations-service.js

```diff
@@ -37,17 +38,59 @@ export function resetTwilioAttributes() {
 export function fetchTwilioAttributes() {
     const {user: {clients, pendingClients}} = store.getState();
     const client = clients.find((client) => client.platform === 'twilio');
-    const pendingClient = pendingClients.find((client) => client.platform === 'twilio');
+    const pendingClient = pendingClients && pendingClients.find((client) => client.platform === 'twilio');
 
     if (client) {
         updateTwilioAttributes({
             linkState: 'linked',
-            number: client.displayName
+            appUserNumber: client.displayName
         });
     } else if (pendingClient) {
         updateTwilioAttributes({
             linkState: 'pending',
-            number: pendingClient.displayName
+            appUserNumber: pendingClient.displayName
         });
     }
 }
+
+export function linkTwilioChannel(userId, data) {
+    return core().appUsers.link.linkChannel(userId, data)
+        .then((appUser) => {
+            store.dispatch(updateUser(...appUser));
```

Comment:
not sure you really need to copy the object here, the reducer should always be doing that